shared/src/uniswap_solver.rs

```diff
@@ -0,0 +1,238 @@
+use ethcontract::{H160, U256};
+use model::TokenPair;
+use std::collections::{HashMap, HashSet};
+
+use crate::uniswap_pool::Pool;
+
+type PathCandidate = Vec<TokenPair>;
+
+pub fn estimate_buy_amount(
+    sell_token: H160,
+    sell_amount: U256,
+    path: &[TokenPair],
```

Comment:
Yes, it's possible. It was mainly because `path_candidates` returned a vector of pairs (to make them easily fetchible). But we can make the conversion `tokens_list_to_pair_list` public and move it into the caller where needed.